packages/core/src/storage/vectorStore/TiDBVectorStore.ts

```diff
@@ -0,0 +1,322 @@
+import type mysql from "mysql2/promise";
+import type { PoolOptions, RowDataPacket } from "mysql2/promise";
+
+import type {
+  VectorStore,
+  VectorStoreQuery,
+  VectorStoreQueryResult,
+} from "./types.js";
+
+import type { GenericFileSystem } from "@llamaindex/env";
+import type { BaseNode, Metadata } from "../../Node.js";
+import { Document, MetadataMode } from "../../Node.js";
+
+export const TIDB_VECTOR_TABLE = "llamaindex_embedding";
+
+interface DocumentEmbedding extends RowDataPacket {
+  id: string;
+  document: string;
+  metadata: Metadata;
+  embeddings: number[];
+  score: number;
+}
+
+/**
+ * Provides support for writing and querying vector data in TiDB.
+ */
+export class TiDBVectorStore implements VectorStore {
+  storesText: boolean = true;
+
+  private namespace: string = "";
+  private tableName: string = TIDB_VECTOR_TABLE;
+  private poolOptions: PoolOptions = {};
+  private dimensions: number = 1536;
+
+  private db?: mysql.Pool;
+
+  /**
+   * Constructs a new instance of the TiDBVectorStore
+   *
+   * @param {object} config - The configuration settings for the instance.
+   * @param {string} config.tableName - The name of the table (optional). Defaults to TIDB_VECTOR_TABLE.
+   * @param {number} config.dimensions - The dimensions of the embedding model.
+   * @param {string} config.poolOptions - The pool options for the TiDB connection.
+   */
+  constructor(config?: {
+    namespace?: string;
+    tableName?: string;
+    dimensions?: number;
+    poolOptions?: PoolOptions;
+    client?: mysql.Pool;
+  }) {
+    this.tableName = config?.tableName ?? TIDB_VECTOR_TABLE;
+    this.namespace = config?.namespace ?? "";
+    this.dimensions = config?.dimensions ?? 1536;
+    this.poolOptions = config?.poolOptions ?? {};
```

Comment:
you can use `Object.assign` to copy the config to this (then you don't need to duplicate your default values)